components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java

```diff
@@ -45,12 +45,23 @@
 	 */
 	public void invokeBusinessLogic(MessageContext msgContext) throws AxisFault {
 		try {
+			if (log.isDebugEnabled()) {
+				log.debug("Request received to DSS.  Data Service :" + msgContext.getServiceContext().getName() + ", " +
+				          "Operation :" + msgContext.getSoapAction() + ", Request body: " +
+				          msgContext.getEnvelope().toString() + ", ThreadID:" + Thread.currentThread().getId());
+			}
 			DataServiceProcessor.dispatch(msgContext);
 		} catch(Exception e) {	
 		    e.printStackTrace();
 			log.error("Error in in-only message receiver", e);
 			msgContext.setProperty(Constants.FAULT_NAME, DBConstants.DS_FAULT_NAME);
 			throw DBUtils.createAxisFault(e);
+		} finally {
+			if (log.isDebugEnabled()) {
+				log.debug("Request processing completed from DSS. Data Service :" +
+				          msgContext.getServiceContext().getName() + ", Operation :" + msgContext.getSoapAction() +
+				          ", ThreadID :" + Thread.currentThread().getId());
```

Comment:
When logging, the format should be "some_message: " + ... notice the colon, and the space after that. This must be consistent, fix in all the places.